Wrappers/Python/ccpi/io/TXRMDataReader.py

```diff
@@ -0,0 +1,203 @@
+        if not(os.path.isfile(self.txrm_file)):
+            raise Exception('File\n {}\n does not exist.'.format(self.txrm_file))  
+
+    def get_geometry(self):
+        
+        '''
+        Return AcquisitionGeometry object
+        '''
+        
+        return self._ag
+        
+    def load_projections(self):
+        
+        '''
+        Load projections and return AcquisitionData container
+        '''
+        
+        # Load projections and most metadata
+        data, metadata = dxchange.read_txrm(self.txrm_file)
+        number_of_images = data.shape[0]
+        
+        # Read source to center and detector to center distances
+        ole = olefile.OleFileIO(self.txrm_file)
+        StoCdist = dxchange.reader._read_ole_arr(ole, \
+                'ImageInfo/StoRADistance', "<{0}f".format(number_of_images))
+        DtoCdist = dxchange.reader._read_ole_arr(ole, \
+                'ImageInfo/DtoRADistance', "<{0}f".format(number_of_images))
+        ole.close()
+        StoCdist = numpy.abs(StoCdist[0])
+        DtoCdist = numpy.abs(DtoCdist[0])
+        
+        # normalise data by flatfield
+        data = data / metadata['reference']
+        
+        # circularly shift data by rounded x and y shifts
+        for k in range(number_of_images):
+            data[k,:,:] = numpy.roll(data[k,:,:], \
+                (int(metadata['x-shifts'][k]),int(metadata['y-shifts'][k])), \
+                axis=(1,0))
+        
+        # Pixelsize loaded in metadata is really the voxel size in um.
+        # We can compute the effective detector pixel size as the geometric
+        # magnification times the voxel size.
+        d_pixel_size = ((StoCdist+DtoCdist)/StoCdist)*metadata['pixel_size']
+        
+        self._ag = AcquisitionGeometry(geom_type = 'cone', 
+                                       dimension = '3D', 
+                                       angles = numpy.degrees(metadata['thetas']), 
+                                       pixel_num_h = metadata['image_width'], 
+                                       pixel_size_h = d_pixel_size/1000, 
+                                       pixel_num_v = metadata['image_height'], 
+                                       pixel_size_v = d_pixel_size/1000, 
+                                       dist_source_center =  StoCdist, 
+                                       dist_center_detector = DtoCdist, 
+                                       channels = 1,
+                                       angle_unit = 'degree')
+
+        return AcquisitionData(array = data, 
+                               deep_copy = False,
+                               geometry = self._ag,
+                               dimension_labels = ['angle', \
+                                                   'vertical', \
+                                                   'horizontal'])
+                
+if __name__ == '__main__':
+    
+    from ccpi.framework import ImageGeometry
```

Comment:
The following must go into a `unittest`.
I think I found the dataset [here ](https://www.fips.fi/dataset.php#walnut)
Possibly we deposit a reconstruction on the CCPi website to test if things work.